File: lib/config/environment.dart
/// Environment configuration for PODSafe
/// 
/// Controls which Firebase project and features are active based on build mode.
library;

enum Environment {
  development,
  production,
}

class EnvironmentConfig {
  // Change this to switch environments
  // DEV: For daily development work
  // PROD: For production releases only
  static const Environment current = Environment.development;
  
  // Environment checks
  static bool get isDevelopment => current == Environment.development;
  static bool get isProduction => current == Environment.production;
  
  // Logging configuration
  static bool get enableVerboseLogging => isDevelopment;
  static bool get enableDebugLogging => isDevelopment;
  
  // Feature flags
  static bool get enableDebugTools => isDevelopment;
  static bool get showEnvironmentBanner => true; // Always show for safety
  
  // Error reporting
  static bool get enableCrashlytics => isProduction;
  static bool get reportErrorsToFirebase => isProduction;
  static bool get enablePrintStatements => isDevelopment; // Control print statements
  
  // Analytics
  static bool get enableAnalytics => true; // Always on, but more detailed in prod
  static bool get collectDetailedAnalytics => isProduction;
  
  // Security
  static bool get enforceStrictSecurity => isProduction;
  
  // Display names
  static String get environmentName {
    switch (current) {
      case Environment.development:
        return 'DEVELOPMENT';
      case Environment.production:
        return 'PRODUCTION';
    }
  }
  
  // Firebase project IDs
  static String get firebaseProjectId {
    switch (current) {
      case Environment.development:
        return 'podsafe-92a3e';
      case Environment.production:
        return 'podsafe-production';
    }
  }
  
  // Public POD viewing base URL
  static String get publicPodBaseUrl {
    switch (current) {
      case Environment.development:
        // For development, use localhost with web port and hash routing
        return 'http://localhost:5000/#';
      case Environment.production:
        return 'https://podsafe.app';
    }
  }
}

File: mobile-rn/src/config/environment.ts
import Config from 'react-native-config';

/**
 * Replaces lib/config/environment.dart's hardcoded Environment.current constant.
 * Driven by react-native-config (.env.development / .env.production, selected via
 * native build flavor/scheme — see vault note "11 Environment and Native Setup").
 *
 * Unlike the Flutter app today (which compiles BOTH dev and prod Firebase credentials
 * into every build regardless of environment), only the active environment's values
 * are present here, since the .env files are not bundled together.
 */
export type Environment = 'development' | 'production';

const environment = (Config.ENVIRONMENT as Environment) ?? 'development';

export const EnvironmentConfig = {
  current: environment,
  isDevelopment: environment === 'development',
  isProduction: environment === 'production',

  enableVerboseLogging: environment === 'development',
  enableDebugLogging: environment === 'development',
  enableDebugTools: environment === 'development',
  showEnvironmentBanner: Config.SHOW_ENVIRONMENT_BANNER === 'true',

  enableCrashlytics: Config.ENABLE_CRASHLYTICS === 'true',
  reportErrorsToFirebase: environment === 'production',
  enableAnalytics: true,
  collectDetailedAnalytics: environment === 'production',

  enforceStrictSecurity: environment === 'production',

  environmentName: environment === 'production' ? 'PRODUCTION' : 'DEVELOPMENT',
  firebaseProjectId: Config.FIREBASE_PROJECT_ID ?? 'podsafe-92a3e',
  publicPodBaseUrl: Config.PUBLIC_POD_BASE_URL ?? 'http://localhost:5000',
};
